src/libsyntax/ext/expand.rs

```diff
@@ -37,11 +38,15 @@ use std::collections::HashSet;
 
 pub fn expand_expr(e: P<ast::Expr>, fld: &mut MacroExpander) -> P<ast::Expr> {
     let expr_span = e.span;
-    return e.and_then(|ast::Expr {id, node, span}| match node {
+    return e.and_then(|ast::Expr {id, node, span, attrs}| match node {
 
         // expr_mac should really be expr_ext or something; it's the
         // entry-point for all syntax extensions.
         ast::ExprMac(mac) => {
+
+            // FIXME: for now, drop attributes on the macro itself
```

Comment:
Not really, this is just to raise awareness about the fact that we drop a piece of the ast on the floor here, and to catch consistency bugs where maybe the attributes get used after all in some of the code following that line.